packages/tiny-graphql-query-compiler/src/index.ts

```diff
@@ -0,0 +1,140 @@
+export interface FieldSelection {
+  [key: string]: boolean | null | undefined | FieldCall | FieldSelection;
+}
+
+export interface BuilderOperation {
+  type: "query" | "subscription" | "mutation";
+  fields: FieldSelection;
+  name?: string;
+  directives?: string[];
+}
+
+const indent = (str: string, depth: number) => {
+  return str
+    .split("\n")
+    .map((line) => " ".repeat(depth) + line)
+    .join("\n");
+};
+
+const compileFieldSelection = (fields: FieldSelection, onlyPresentVariableValues = false): string => {
+  const result = Object.entries(fields)
+    .map(([field, value]) => {
+      if (typeof value === "boolean") {
+        return value ? field : false;
+      } else if (value instanceof FieldCall) {
+        let args = "";
+        let entries = Object.entries(value.args).filter(([_, value]) => value !== null && value !== undefined);
+
+        if (onlyPresentVariableValues) {
+          entries = entries.filter(([_, value]) => (value instanceof Variable ? value.present() : true));
```

Comment:
naming nit: I was actually reading `value.present()` as "present this variable" vs "is present" (which I figured out after checking out the class). A little room for confusion, so up to you if you'd like to maybe clarify that it's the latter!